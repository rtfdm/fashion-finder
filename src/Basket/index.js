import React, { Component, Fragment } from 'react'
import styled from 'styled-components'
import { ArrowLeft, ShoppingBag } from 'react-feather'
import { Link } from 'react-router-dom'
import basketItems from './sampleItems'
import Item from '../Item/index'

const BackButton = ({ IconComponent, className }) => (
  <Link to="/" id="back-button" className={className}>
    <div>{<IconComponent size="24" color="#838383" />}</div>
    <div>Back</div>
  </Link>
)

const StyledBackButton = styled(BackButton)`
  display: grid;
  grid-template-columns: min-content min-content;
  grid-column-gap: 10px;
  width: min-content;
  align-items: center;
  font-size: 18px;
  color: #838383;
  margin-top: -35px;
  margin: 10px 0;
  transition: all ease 0.5s;
  text-decoration: none;

  @media only screen and (max-width: 800px) {
    margin: 0;
    padding: 20px 10px;
  }
`

const LookPageLink = ({ className, id, IconComponent, color }) => {
  return (
    <Link to="/">
      <div className={className} id={id}>
        {<IconComponent color={color} size="20" style={ArrowMargin} />}
        <br />
        Get inspired
      </div>
    </Link>
  )
}

const Container = styled.div`
  max-width: 1200px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
`

const ListContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  align-items: stretch;
  justify-items: center;
  grid-template-rows: auto;
  grid-gap: 40px;
  max-width: 1000px;
  background-color: #fff;
  padding: 0;
  margin: 0;
  list-style: none;
  margin-bottom: 120px;

  @media only screen and (max-width: 599px) {
    margin: 0 auto;
    grid-gap: 0px;
  }

  @media only screen and (max-width: 1029px) {
    grid-gap: 40px;
    justify-items: space-evenly;
  }
`

const HeaderStyle = styled.div`
  max-width: 1200px;
  background-color: #fff;
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  margin: 0;
  list-style: none;
`
const FooterStyling = styled.div`
  display: grid;
  grid-template-columns: repeat(11, 2fr);
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  height: 70px;
  opacity: 0.8;
  padding: 10px;
  color: #ffff
  text-align: center;
  min-width: 80vw;
`

const BuyButton = ({ className, id, IconComponent, color }) => {
  return (
    <div className={className} id={id}>
      {<IconComponent color={color} size="32" />}
    </div>
  )
}

const StyledBuyButton = styled(BuyButton)`
  grid-column: 6;
  background-color: #a6befa;
  box-shadow: 0px 4px 20px rgba(100, 100, 100, 100);
  border-radius: 100px;
  max-height: 62px;
  padding: 10px;
  margin: 0 auto;
`

const ArrowMargin = {
  marginLeft: '25px',
}

class Basket extends Component {
  _isBasketEmpty(object) {
    if (Object.keys(object).length === 0) return true
    return false
  }

  renderBasket() {
    if (this._isBasketEmpty(this.props.basket)) {
      return <h2>You don't have any basket items</h2>
    } else {
      return Object.keys(this.props.basket).map(key => (
        <Item key={key} details={this.props.basket[key]} />
      ))
    }
  }

  render() {
    return (
      <Container>
        <HeaderStyle>
          <StyledBackButton IconComponent={ArrowLeft} />
        </HeaderStyle>
        <HeaderStyle>
          <h1>Your Lookbook</h1>
        </HeaderStyle>
        <ul>
          <ListContainer>{this.renderBasket()}</ListContainer>
        </ul>
        <FooterStyling>
          <StyledBuyButton
            id="buy-button"
            IconComponent={ShoppingBag}
            color="#ffff"
          />
        </FooterStyling>
      </Container>
    )
  }
}

export default Basket
