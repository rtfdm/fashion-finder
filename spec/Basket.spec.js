import React from 'react'
import renderer from 'react-test-renderer'
import Basket from '../src/Basket'
import { exportAllDeclaration } from '@babel/types'
import { JestEnvironment } from '@jest/environment'
require('react-router-dom')
jest.mock('react-router-dom')

let basketItems

beforeEach(() => {
  basketItems = {
    id: 1,
    brand: 'J',
    desc: 'J is awesome',
    price: 3000,
    image: 'https://i.imgur.com/iKT9fl6.jpg',
    gender: 'M',
  }
})

test('Basket', () => {
  const renderedComponent = renderer
    .create(<Basket basketItems={basketItems} />)
    .toJSON()
  expect(renderedComponent).toMatchSnapshot()
})
