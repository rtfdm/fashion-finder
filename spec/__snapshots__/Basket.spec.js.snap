// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket 1`] = `
<div
  className="sc-ifAKCX gtuAff"
>
  <div
    className="sc-bZQynM evpxTC"
  />
  <div
    className="sc-bZQynM evpxTC"
  >
    <h1>
      Your Lookbook
    </h1>
  </div>
  <ul>
    <div
      className="sc-EHOje bnCEOK"
    >
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="White-tailed deer"
          height="300px"
          src="https://source.unsplash.com/300x300/?mountain"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          White-tailed deer
        </h3>
        <p>
          Hello hello this is an example
        </p>
        <p>
          Price: 
          $3.45
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="White-tailed deer"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="White-tailed deer"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="European badger"
          height="300px"
          src="https://source.unsplash.com/300x300/?house"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          European badger
        </h3>
        <p>
          Such a cool shirt
        </p>
        <p>
          Price: 
          $0.25
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="European badger"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="European badger"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="Loris is a really nice person who is a senior developer"
          height="300px"
          src="https://source.unsplash.com/300x300/?fashion,clothing"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          Loris is a really nice person who is a senior developer
        </h3>
        <p>
          The year is 1987 and NASA launches the last of America's deep space probes. In a freak mishap, Ranger 3 and its pilot Captain William     'Buck' Rogers are blown out of their trajectory into an orbit which freezes     his life support system and returns Buck Rogers to Earth five hundred years later.
        </p>
        <p>
          Price: 
          $2.82
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="Loris is a really nice person who is a senior developer"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="Loris is a really nice person who is a senior developer"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="White-browed owl"
          height="300px"
          src="https://source.unsplash.com/300x300/?dog"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          White-browed owl
        </h3>
        <p>
          -JKLASDJKASKDJAKL:SJDjajsdjajsdjasd
        </p>
        <p>
          Price: 
          $1.30
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="White-browed owl"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="White-browed owl"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="Coatimundi, ring-tailed"
          height="300px"
          src="https://source.unsplash.com/300x300/?bicycles\`"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          Coatimundi, ring-tailed
        </h3>
        <p>
          LOrem Ipsum yo
        </p>
        <p>
          Price: 
          $8.30
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="Coatimundi, ring-tailed"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="Coatimundi, ring-tailed"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="Black swan"
          height="300px"
          src="https://source.unsplash.com/300x300/?rocks"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          Black swan
        </h3>
        <p>
          :D :D :D :D :D :)
        </p>
        <p>
          Price: 
          $7.82
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="Black swan"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="Black swan"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="Test"
          height="300px"
          src="https://source.unsplash.com/300x300/?videogame"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          Test
        </h3>
        <p>
          Hello hello
        </p>
        <p>
          Price: 
          $7.82
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="Test"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="Test"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
      <li
        className="sc-bdVaJa ikFPsF"
      >
        <img
          alt="Test"
          height="300px"
          src="https://source.unsplash.com/300x300/?vintage"
          style={
            Object {
              "borderRadius": "5px",
              "maxWidth": "100%",
            }
          }
          width="300px"
        />
        <h3>
          Test
        </h3>
        <p>
          Hello hello
        </p>
        <p>
          Price: 
          $7.82
        </p>
        <div
          className="sc-bwzfXH jLfHiB"
          id="Test"
        >
          <svg
            fill="none"
            height="32"
            stroke="#EE8CA3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
        <div
          className="sc-htpNat ipDBtf"
          id="Test"
        >
          <svg
            fill="none"
            height="32"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </div>
      </li>
    </div>
  </ul>
  <div
    className="sc-gzVnrw kMOvrt"
  >
    <div
      className="sc-htoDjs kPtwwD"
      id="buy-button"
    >
      <svg
        fill="none"
        height="32"
        stroke="#ffff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
        />
      </svg>
    </div>
  </div>
</div>
`;
